### ISSUE TITLE
add events once the app is loaded

### ISSUE THREAD

**Issue by AntouanK** [mikecao/umami#398]
Hi there.

Thanks for your amazing work.

Quick question.
I have a SPA and I load the umami tracker. The problem is that even if I have the classes in the correct elements, umami doesn't see them because it looks for them before they exist.

More specifically the browser ( in time order ) : 
- loads the app js
- runs the app js
- loads the umami tracker
- the umami tracker runs `addEvents` ( https://github.com/mikecao/umami/blob/master/tracker/index.js#L171 ) but finds nothing ( app is not done yet )
- the app is now done and the DOM has the elements with the classes ( oops, too late )

Is there a way to run `addEvents` when my app is done?
As far as I can see that function is not exposed.
Am I missing something?

Thank you.

**Comment by mikecao** [comment]
The `defer` attribute should make the script load after the page. If that doesn't work, you can try dynamically writing the script tag after your app completely loads.

I'll look into exposes the event methods. I'm just concerned about potential duplicate stats. But I guess if you're calling it manually you know the risks.

**Comment by AntouanK** [comment]
yes, I ended up adding the script after my app has loaded.

I looked a bit into that tracker code. 
You're right, I guess running `addEvents` multiple times will add duplicate listeners to elements potentially.

Could we maybe map them as we add them?
So in any subsequent call to `addEvents` we can use that map to clear the existing element listeners first?

Otherwise, I'm thinking that a proper solution might be for my app to add custom events, and not use the classes.
So that offloads the work to my app and not umami ( since the way to add custom events is already available )

btw, I'm using this for my HN reader --> https://hack.ernews.info/
code for the front-end is here https://gitlab.com/antouank/hni-ui ( the whole project is open source )
and the way I did it for now is this https://gitlab.com/antouank/hni-ui/-/blob/master/staticAssets/main.js#L108
but it's very hacky, so I want to have a proper solution.

Let me know what you think.

Thank you.
